pkgs/stdenv/generic/setup.sh

```diff
@@ -212,6 +212,18 @@ isELF() {
     if [ "$magic" = $'\177ELF' ]; then return 0; else return 1; fi
 }
 
+# Return success if the specified file is an ELF object
+# and its e_type is ET_EXEC (executable file)
+isELFExec() {
+    grep -ao -P '^\177ELF.{11}\x00\x02' "$1" >/dev/null
```

Comment:
That one might be possible. Could be over-engineering though.

```suggestion
    isELF "$1" && grep -ao -P '^\177ELF.{11}\x00\x02' "$1" >/dev/null
```

Reply:
It's already only being run on executable files. Is the extra complexity really worth it?
If so I want to see a comment explaining _why_ the implementation is the way it is.

Reply:
It's currently only used in the `wrapQtApps` hook, and there only run on files with the executable bit set.

I doubt we would even be able to measure the difference if we added that optimization into stdenv.
I'd be happy to sacrifice a non-measurable performance difference for better readability.

Reply:
fair enough.